prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -189,6 +189,11 @@ class ReportFunction(
                     ActionLog(InvalidReportMessage(e.message ?: "Invalid request."), type = ActionLogLevel.error)
                 )
                 HttpStatus.BAD_REQUEST
+            } catch (e: Options.Companion.InvalidOptionException) {
```

Comment:
I think this replaces the IllegalArgumentException and it could be removed.

Reply:
So the reason I added this is that the scope for the `IllegalArgumentException` is `report` but adding this additional catch, I can specify `parameters` as the scope of the exception.